doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -1,394 +1,648 @@
+    visual_tools.publishText(text_pose, "Cartesian Planning", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+
+    // Set the target Pose to our current location
+    target_pose.orientation.w = 0.0;
+    target_pose.orientation.x = 0.383;
+    target_pose.orientation.y = -0.9240;
+    target_pose.orientation.z = -0.0;
+    target_pose.position.x = 0.28;
+    target_pose.position.y = -0.2;
+    target_pose.position.z = 0.5;
+
+    // BEGIN_SUB_TUTORIAL cartesian_path
+    // Cartesian Paths
+    // ^^^^^^^^^^^^^^^
+    //
+    // You can plan a Cartesian path directly by specifying a list of waypoints
+    // for the end-effector to go through. Note The initial pose (target_pose) does not
+    // need to be added to the waypoint list but adding it can help with visualizations
+    std::vector<geometry_msgs::Pose> waypoints;
+    waypoints.push_back(target_pose);
+
+    target_pose.position.z -= 0.2;
+    waypoints.push_back(target_pose);  // down
+
+    target_pose.position.y -= 0.2;
+    waypoints.push_back(target_pose);  // right
+
+    target_pose.position.z += 0.2;
+    target_pose.position.y += 0.2;
+    target_pose.position.x -= 0.2;
+    waypoints.push_back(target_pose);  // up, left, and back
+
+    // Cartesian motions are frequently needed to be slower for actions such as approach and
+    // retreat grasp motions. Here we demonstrate how to reduce the speed of the robot arm via
+    // a scaling factor of the maxiumum speed of each joint. Note this is not the speed of the
+    // end effector point.
+    move_group.setMaxVelocityScalingFactor(0.1);
+
+    // We want the Cartesian path to be interpolated at a resolution of 1 cm
+    // which is why we will specify 0.01 as the max step in Cartesian
+    // translation.  We will specify the jump threshold as 0.0, effectively disabling it.
+    // Warning - disabling the jump threshold while operating real hardware can cause
+    // large unpredictable motions of redundant joints and could be a safety issue.
+    moveit_msgs::RobotTrajectory trajectory;
+    const double jump_threshold = 0.0;
+    const double eef_step = 0.01;
+    double fraction = move_group.computeCartesianPath(waypoints, eef_step, jump_threshold, trajectory);
+    ROS_INFO_NAMED("tutorial", "Visualizing plan 5 (Cartesian path) (%.2f%% acheived)", fraction * 100.0);
+
+    // While we didn't use the planner to create this trajectory it still can be added to a plan
+    my_plan.trajectory_ = trajectory;
+
+    // END_SUB_TUTORIAL
+
+    // Visualize the plan in RViz, becuase there are multiple points we use a loop to show them all
+    visual_tools.publishPath(waypoints, rvt::LIME_GREEN, rvt::SMALL);
+    for (std::size_t i = 0; i < waypoints.size(); ++i)
+      visual_tools.publishAxisLabeled(waypoints[i], "pt" + std::to_string(i), rvt::SMALL);
+    visual_tools.trigger();
+    visual_tools.prompt("Press 'next' to move along the cartesian path");
+  }
+
+  void create_obstacle()
```

Comment:
this should take in the size and pose of the object. As it stands, this doesn't demonstrate re-usable code. It is essentially a script broken up into clumps.